driver-core/src/test/unit/com/mongodb/connection/DefaultServerSpecification.groovy

```diff
@@ -146,36 +149,55 @@ class DefaultServerSpecification extends Specification {
         server.getConnection()
 
         then:
-        thrown(MongoSecurityException)
+        def e = thrown(MongoException)
+        e.is(exceptionToThrow)
         1 * connectionPool.invalidate()
         1 * serverMonitor.connect()
+
+        where:
+        exceptionToThrow << [
+                new MongoSecurityException(createCredential('jeff', 'admin', '123'.toCharArray()), 'Auth failed'),
+                new MongoSocketOpenException('open failed', new ServerAddress(), new IOException()),
+                new MongoSocketWriteException('Write failed', new ServerAddress(), new IOException()),
+                new MongoSocketReadException('Read failed', new ServerAddress(), new IOException()),
+                new MongoSocketReadTimeoutException('Read timed out', new ServerAddress(), new IOException()),
+        ]
     }
 
-    def 'failed open should invalidate the server asychronously'() {
+    def 'failed open should invalidate the server asynchronously'() {
         given:
         def clusterTime = new ClusterClock()
+        def connectionPool = Mock(ConnectionPool)
```

Comment:
No reason not to use a Mock here as well, and then we can check that invalidate is called.